raw/fetch_test.py

```diff
@@ -0,0 +1,164 @@
+import configparser
+import datetime
+import json
+import os
+import tempfile
+import unittest
+
+from unittest.mock import patch, MagicMock
+
+# with . or not below??
+from fetch import (
+    new_config_file,
+    get_month_range,
+    get_single_level_dates,
+    get_previous_month_dates,
+    update_config_files,
+    get_secret,
+)
+
+class TestFetchFunctions(unittest.TestCase):
+    def setUp(self):
+        # Create a temporary directory for testing
+        self.temp_dir = tempfile.mkdtemp()
+        self.config_file = os.path.join(self.temp_dir, "test_config.cfg")
+        with open(self.config_file, "w") as file:
+            file.write(
+                "[parameters]\nclient=cds\ndataset=reanalysis-era5-complete\n\
+                target_path=gs://gcp-public-data-arco-era5/raw/ERA5GRIB/HRES\
+                /Daily/{date:%%Y/%%Y%%m%d}_hres_dve.grb2\npartition_keys=\n\t\
+                dates\n\n[selection]\nclass=ea\nstream=oper\nexpver=1\ntype=an\n\
+                levtype=ml\nlevelist=1/to/137\ndate=1979-01-01/to/2023-03-31\n\
+                time=00/to/23\nparam=138/155\n"
+            )
+        self.first_day_first_prev = datetime.date(2023, 7, 1)
+        self.last_day_first_prev = datetime.date(2023, 7, 31)
+        self.first_day_third_prev = datetime.date(2023, 5, 1)
+        self.last_day_third_prev = datetime.date(2023, 5, 31)
+        self.sl_year, self.sl_month = "2023", "05"
+        self.sl_first_date, self.sl_last_date = "01", "31"
+        self.additional_content = "[parameters.test]\napi_url=test_url\napi_key=\
+            test_key\n\n"
+
+    def tearDown(self):
+        os.remove(self.config_file)
+        os.rmdir(self.temp_dir)
```

Comment:
```suggestion
        self.temp_dir.cleanup()
```

See https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory